# PRUEBAS


Probando los diferentes conceptos que voy aprendiendo durante la competición sobre ML o DS

In [15]:
import pandas as pd

---

<p align="center">
  <img src="separador.png" alt="Separador" width=900"/>
</p>

---


# 1 | Exploración de Datos (EDA)

Primera etapa. Este cuadernillo irá abordando las diferentes etapas y en el orden en el que se deberían realizar.

In [2]:
train = pd.read_csv("train.csv")

In [4]:
train.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

In [8]:
train.shape
# Mirando las dimensiones del DataFrame, al cual hemos llamado "train". Tiene 891 filas x 12 columnas

(891, 12)

In [10]:
train.columns
#Asi mostramos el nombre de las 12 columnas

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

### Razonamiento:

Podemos decir que en el dataframe "train" habrá columnas de relativa importancia para la predicción, y otras no. Opinión personal es que *Name*, *Ticket*, *Cabin* y *Embarked* no son de demasiada utilidad.

In [11]:
# Vemos los primeros 5 registros del dataframe. Para los 5 ultimos , usado anteriormente, sería .tail()
train.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

### Los modelos no entienden valores de TEXTO (inicialmente)

Ejemplos: 

    *Name*, *Sex*, *Ticket*, *Cabin*, *Embarked*

Habrá que ***transformarlos*** o ***descartarlos***


El 80% del trabajo en ML real es identificar los ***3 grandes problemas clásicos***:

-Valores nulos

-Variables categóricas (ejemplo *Sex*:***Male|Female**)

-Texto Libre

In [12]:
train.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

## ¿Cuál es el objetivo de esta etapa ?

La **Exploración de Datos (EDA)** no es para entrenar modelos.  
Es para **no cometer errores tontos después**.

 El objetivo es:

> **Entender qué tipo de datos tenemos y si tienen sentido**

Nada más. Nada menos.



---

<p align="center">
  <img src="separador.png" alt="Separador" width=1000"/>
</p>

---



---

# 2 | Resumen estadístico</h1>


## ¿Por qué mirar la “distribución” de los números?

Cuando digo *distribución*, no pienses en fórmulas.  
Piensa en responder preguntas simples como:

- ¿Los valores son pequeños o grandes?
- ¿Hay cosas absurdas?
- ¿Hay valores que faltan?
- ¿Todo está en el rango esperado?

---

## Ejemplo fuera del Titanic

Imagina que tienes una columna **Edad**:

- Edad mínima = `-5`
- Edad máxima = `300`

Aunque no sepas estadística, sabes que algo esta mal.


EDA sirve para detectar eso **antes de entrenar nada**.

---

## Ejemplos concretos en Titanic

### Age
Queremos saber:
- ¿Hay edades negativas?
- ¿Hay gente de 200 años?
- ¿Faltan edades?

---

### Fare (precio)
Queremos ver:
- ¿Hay tickets gratis?
- ¿Hay precios extremadamente altos?
- ¿La mayoría pagó poco o mucho?

---

### SibSp y Parch
Queremos ver:
- ¿Viaja la gente sola?
- ¿Con familias grandes?
- ¿Hay valores rarísimos?

---

## ¿Qué pasa si NO hacemos esta etapa?

Sin EDA:
- El modelo se confunde
- Aprende cosas incorrectas
- Tus resultados empeoran
- Y no sabes por qué 





---


---

In [14]:
train.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

***Observaciones*** :

-Podemos ver que la edad minima es 0.42. Aunque este dato en principio parezca que esta mal, había bebes a bordo. El dato es correcto.

-*Age* tiene muchos valores faltantes. De 891 pasajeros tenemos el valor *Age* de 714.



---

<p align="center">
  <img src="separador.png" alt="Separador" width=1000"/>
</p>

---


# 3 | Limpieza de datos (Data Cleaning)